File: avalon-modules/avalon-authorization/src/main/java/io/github/lishangbu/avalon/authorization/service/impl/VerificationCodeServiceImpl.java
package io.github.lishangbu.avalon.authorization.service.impl;

import io.github.lishangbu.avalon.oauth2.common.properties.Oauth2Properties;
import lombok.extern.slf4j.Slf4j;
import org.springframework.data.redis.core.StringRedisTemplate;
import org.springframework.stereotype.Service;

/// 验证码服务实现
///
/// 使用 Redis 存储验证码，并通过日志输出短信与邮件内容
///
/// @author lishangbu
/// @since 2026/3/13
@Slf4j
@Service
public class VerificationCodeServiceImpl extends AbstractVerificationCodeService {

    public VerificationCodeServiceImpl(
            StringRedisTemplate stringRedisTemplate, Oauth2Properties oauth2Properties) {
        super(stringRedisTemplate, oauth2Properties);
    }

    @Override
    protected void deliverCode(String type, String target, String code) {
        log.info("发送验证码 -> type={}, target={}, code={}", type, target, code);
    }
}

File: avalon-modules/avalon-authorization/src/main/kotlin/io/github/lishangbu/avalon/authorization/service/impl/VerificationCodeServiceImpl.kt
package io.github.lishangbu.avalon.authorization.service.impl

import io.github.lishangbu.avalon.oauth2.common.properties.Oauth2Properties
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.data.redis.core.StringRedisTemplate
import org.springframework.stereotype.Service

/**
 * 验证码服务实现
 *
 * 使用 Redis 存储验证码，并通过日志输出短信与邮件内容
 *
 * @author lishangbu
 * @since 2026/3/13
 */
@Service
class VerificationCodeServiceImpl(
    stringRedisTemplate: StringRedisTemplate,
    oauth2Properties: Oauth2Properties,
) : AbstractVerificationCodeService(stringRedisTemplate, oauth2Properties) {
    override fun deliverCode(
        type: String,
        target: String,
        code: String,
    ) {
        log.info("发送验证码 -> type={}, target={}, code={}", type, target, code)
    }

    companion object {
        private val log: Logger = LoggerFactory.getLogger(VerificationCodeServiceImpl::class.java)
    }
}
